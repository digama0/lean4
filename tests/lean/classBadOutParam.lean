
class C1 (x : OutParam Nat) (y : { n : Nat // n > x }) (α : Type) := -- should fail
(val : α)

class C2 (x : OutParam Nat) (y : OutParam { n : Nat // n > x }) (α : Type) := -- should work
(val : α)
