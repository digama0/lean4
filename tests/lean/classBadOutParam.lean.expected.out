classBadOutParam.lean:3:1-3:4: error: invalid class, parameter #2 depends on `OutParam`, but it is not an `OutParam`
